### ISSUE TITLE
Regression: arrow_alignment check broken between 2.1.1 and 2.2.x releases when left side contains text after variable interpolation

### ISSUE THREAD

**Issue by dbeckham** [rodjek/puppet-lint#697]
### Description of the Problem
At some point between version 2.1.1 and the 2.2.x releases, the arrow_alignment check was broken when a string with containing a variable **with characters after the variable** exists on the left side of the arrow (`=>`).

### How to Reproduce

The following manifest was tested on the 2.2.0 and 2.2.1 releases as well as the master branch (as of  8b5ee3d3560e411d4b6f78b2f0df8ebc015d212d). 

*NOTE: if you run puppet-lint with the `--fix` option against this manifest, it will also report each error as fixed, but no changes are actually made.* 

#### Test Manifest
```puppet
class arrow_alignment_check {

    $somevar = 'bar'

    class { 'foo':
        ensure => 'present',
        param  => {
            "${somevar}baz" => true,
            'foo'           => false,
        }
    }

}
```

#### Error Output
```
$ puppet-lint --error-level all --with-context --only-checks arrow_alignment  manifests/arrow_alignment_check.pp

WARNING: indentation of => is not properly aligned (expected in column 29, but found it in column 26) on line 8

  "${somevar}baz" => true,
                           ^
```

### Full Testing Manifest

```
#
# Run with:
#   puppet-lint --error-level all --with-context --only-checks arrow_alignment  manifests/arrow_alignment_check.pp
#
class arrow_alignment_check {

    $somevar = 'bar'

    #
    # These examples DO NOT fail the arrow_alignment check
    #

    # No variable used
    class { 'foo':
        ensure => 'present',
        param  => {
            'foobar' => true,
            'foo'    => false,
        }
    }

    # No characters AFTER the variable
    class { 'foo':
        ensure => 'present',
        param  => {
            "foo${somevar}" => true,
            'foo'           => false,
        }
    }

    # The variable by itlsef
    class { 'foo':
        ensure => 'present',
        param  => {
            "${somevar}" => true,
            'bar'        => false,
        }
    }

    #
    # These examples DO fail in the 2.2.x releases
    #

    # Original example with an interpolated variable
    class { 'foo':
        ensure => 'present',
        param  => {
            "foo${somevar}baz" => true,
            'foo'              => false,
        }
    }

    # No characters BEFORE the variable
    class { 'foo':
        ensure => 'present',
        param  => {
            "${somevar}baz" => true,
            'foo'           => false,
        }
    }

    # The problem occurs at any depth
    class { 'foo':
        ensure => 'present',
        param  => {
            "foo${somevar}baz" => true,
            'foo'              => false,
            'param'            => {
                "foo${somevar}baz" => true,
                'foo'              => false,
                'param'            => {
                    "foo${somevar}baz" => true,
                    'foo'              => false,
                }
            }
        }
    }

}
```

**Comment by dbeckham** [comment]
I'm wondering if the three merges from #657 that reorganized the check layout didn't either introduce the regression from hidden changes merged along with the reorg, or is just missing part of the recent changes made to fix the arrow alignment check problems in version 2.1.1.

**Comment by rnelson0** [comment]
You implied it but just to confirm, you see the same behavior with 2.2.1
gem/release and HEAD?
-- 
Rob Nelson

**Comment by dbeckham** [comment]
Sorry for the late reply.  Yes, I did see the issue in 2.2.1 and HEAD.  But, we have been in the process of upgrading to Puppet 4, and part of that was upgrading our testing environment, gems, and Ruby version.  After the upgrade, I no longer see this issue, but I don't have any good information for you about the environment I was testing this under, other than I know that I was still testing under Ruby 1.9.3 when I submitted this.

I think there is some combination of older gems installed and/or Ruby 1.9.3 that can cause this error to happen, but I couldn't tell you how to reproduce it at this point.  I'm to go ahead and close this as upgrading to reasonably new versions of Ruby and gem dependencies seems to fix the issue.

**Comment by dbeckham** [comment]
Ok, I was wrong earlier. I thought I had updated my puppet-lint version to the latest along with the Puppet 4 fixes, but I had not. I was using 2.1.x.   After updating to the latest puppet-lint version (this repository's HEAD), I'm definitely still seeing this issue.  

If it helps, I've attached my `Gemfile` and `Gemfile.lock` files.

#### Gemfile

```
source 'https://rubygems.org'

ruby '2.1.9'

gem 'rake', '~> 10.0'
gem 'rspec-its', '~> 1.0'
gem 'rspec-collection_matchers', '~> 1.0'
gem 'rspec', '~> 3.0'

gem 'puppet', '~> 4.10.0'
gem 'hiera-eyaml',                        :require => false

gem 'puppet_facts',                       :require => false
gem 'puppetlabs_spec_helper', '>= 1.2.1', :require => false
gem 'rspec-puppet', '>= 2.3.2',           :require => false
gem 'rspec-puppet-facts',                 :require => false
gem 'simplecov',                          :require => false
gem 'parallel_tests',                     :require => false
gem 'fast_gettext',                       :require => false
gem 'json',                               :require => false

gem 'ci_reporter_rspec',                  :require => false

# Puppet Lint Plugins
gem 'puppet-lint', :git => 'https://github.com/rodjek/puppet-lint.git', :require => false

gem 'puppet-lint-absolute_classname-check',                         :require => false
gem 'puppet-lint-absolute_template_path',                           :require => false
gem 'puppet-lint-classes_and_types_beginning_with_digits-check',    :require => false
gem 'puppet-lint-concatenated_template_files-check',                :require => false
gem 'puppet-lint-duplicate_class_parameters-check',                 :require => false
gem 'puppet-lint-file_ensure-check',                                :require => false
gem 'puppet-lint-leading_zero-check',                               :require => false
gem 'puppet-lint-no_symbolic_file_modes-check',                     :require => false
gem 'puppet-lint-param-docs',                                       :require => false
gem 'puppet-lint-resource_reference_syntax',                        :require => false
gem 'puppet-lint-spaceship_operator_without_tag-check',             :require => false
gem 'puppet-lint-trailing_comma-check',                             :require => false
gem 'puppet-lint-undef_in_function-check',                          :require => false
gem 'puppet-lint-unquoted_string-check',                            :require => false
gem 'puppet-lint-variable_contains_upcase',                         :require => false
gem 'puppet-lint-version_comparison-check',                         :require => false

# vim: ts=2 sts=2 sw=2 et ft=ruby:
```

#### Gemfile.lock

```
GIT
  remote: https://github.com/rodjek/puppet-lint.git
  revision: 8b5ee3d3560e411d4b6f78b2f0df8ebc015d212d
  specs:
    puppet-lint (2.2.1)

GEM
  remote: https://rubygems.org/
  specs:
    builder (3.2.3)
    ci_reporter (2.0.0)
      builder (>= 2.1.2)
    ci_reporter_rspec (1.0.0)
      ci_reporter (~> 2.0)
      rspec (>= 2.14, < 4)
    diff-lcs (1.3)
    docile (1.1.5)
    facter (2.4.6)
    facterdb (0.3.10)
      facter
      jgrep
    fast_gettext (1.1.0)
    gettext (3.2.2)
      locale (>= 2.0.5)
      text (>= 1.3.0)
    gettext-setup (0.23)
      fast_gettext (~> 1.1.0)
      gettext (>= 3.0.2)
      locale
    hiera (3.3.1)
    hiera-eyaml (2.1.0)
      highline (~> 1.6.19)
      trollop (~> 2.0)
    highline (1.6.21)
    jgrep (1.4.1)
      json
    json (2.0.4)
    json_pure (1.8.6)
    locale (2.1.2)
    mcollective-client (2.10.3)
      json
      stomp
      systemu
    metaclass (0.0.4)
    mocha (1.2.1)
      metaclass (~> 0.0.1)
    parallel (1.11.1)
    parallel_tests (2.14.1)
      parallel
    puppet (4.10.0)
      facter (> 2.0, < 4)
      gettext-setup (>= 0.10, < 1)
      hiera (>= 2.0, < 4)
      json_pure (~> 1.8)
      locale (~> 2.1)
    puppet-lint-absolute_classname-check (0.2.4)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-absolute_template_path (1.0.1)
      puppet-lint (>= 1.1, < 3.0)
    puppet-lint-classes_and_types_beginning_with_digits-check (0.1.2)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-concatenated_template_files-check (0.1.1)
      puppet-lint (>= 1.1, < 3.0)
    puppet-lint-duplicate_class_parameters-check (1.0.2)
      puppet-lint (>= 1.1, < 3.0)
    puppet-lint-file_ensure-check (0.3.1)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-leading_zero-check (0.1.1)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-no_symbolic_file_modes-check (1.0.2)
      puppet-lint (>= 1.1, < 3.0)
    puppet-lint-param-docs (1.4.2)
      puppet-lint (>= 1.1, < 3.0)
    puppet-lint-resource_reference_syntax (1.0.10)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-spaceship_operator_without_tag-check (0.1.1)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-trailing_comma-check (0.3.2)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-undef_in_function-check (0.2.1)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-unquoted_string-check (0.3.0)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-variable_contains_upcase (1.2.0)
      puppet-lint (>= 1.0, < 3.0)
    puppet-lint-version_comparison-check (0.2.1)
      puppet-lint (>= 1.0, < 3.0)
    puppet-syntax (2.4.0)
      rake
    puppet_facts (0.2.1)
    puppetlabs_spec_helper (2.1.1)
      mocha (~> 1.0)
      parallel_tests
      puppet-lint (~> 2.0)
      puppet-syntax (~> 2.0)
      rspec-puppet (~> 2.0)
    rake (10.5.0)
    rspec (3.5.0)
      rspec-core (~> 3.5.0)
      rspec-expectations (~> 3.5.0)
      rspec-mocks (~> 3.5.0)
    rspec-collection_matchers (1.1.3)
      rspec-expectations (>= 2.99.0.beta1)
    rspec-core (3.5.4)
      rspec-support (~> 3.5.0)
    rspec-expectations (3.5.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.5.0)
    rspec-its (1.2.0)
      rspec-core (>= 3.0.0)
      rspec-expectations (>= 3.0.0)
    rspec-mocks (3.5.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.5.0)
    rspec-puppet (2.5.0)
      rspec
    rspec-puppet-facts (1.7.1)
      facter
      facterdb (>= 0.3.0)
      json
      mcollective-client
      puppet
    rspec-support (3.5.0)
    simplecov (0.14.1)
      docile (~> 1.1.0)
      json (>= 1.8, < 3)
      simplecov-html (~> 0.10.0)
    simplecov-html (0.10.0)
    stomp (1.4.3)
    systemu (2.6.5)
    text (1.3.1)
    trollop (2.1.2)

PLATFORMS
  ruby

DEPENDENCIES
  ci_reporter_rspec
  fast_gettext
  hiera-eyaml
  json
  parallel_tests
  puppet (~> 4.10.0)
  puppet-lint!
  puppet-lint-absolute_classname-check
  puppet-lint-absolute_template_path
  puppet-lint-classes_and_types_beginning_with_digits-check
  puppet-lint-concatenated_template_files-check
  puppet-lint-duplicate_class_parameters-check
  puppet-lint-file_ensure-check
  puppet-lint-leading_zero-check
  puppet-lint-no_symbolic_file_modes-check
  puppet-lint-param-docs
  puppet-lint-resource_reference_syntax
  puppet-lint-spaceship_operator_without_tag-check
  puppet-lint-trailing_comma-check
  puppet-lint-undef_in_function-check
  puppet-lint-unquoted_string-check
  puppet-lint-variable_contains_upcase
  puppet-lint-version_comparison-check
  puppet_facts
  puppetlabs_spec_helper (>= 1.2.1)
  rake (~> 10.0)
  rspec (~> 3.0)
  rspec-collection_matchers (~> 1.0)
  rspec-its (~> 1.0)
  rspec-puppet (>= 2.3.2)
  rspec-puppet-facts
  simplecov

RUBY VERSION
   ruby 2.1.9p490

BUNDLED WITH
   1.14.6
```

**Comment by rodjek** [comment]
Fixed in #701
